### ISSUE TITLE
ElasticNet fails to predict on basic time series

### ISSUE THREAD

**Issue by ParthivNaresh** [alteryx/evalml#2756]
ElasticNet (which is a linear model) fails to predict on basic time series problems. This is the only `linear_model` we leverage in time series regression.

```
import numpy as np
import pandas as pd
from evalml import AutoMLSearch

samples = np.linspace(0, 100, 200, endpoint=False)
samples = np.linspace(-np.pi, np.pi, 201)
signal = np.sin(2 * np.pi * samples)

X = pd.DataFrame({"date": pd.date_range("1/1/2020", periods=200)})
y = pd.Series(signal)

aml = AutoMLSearch(X_train=X.iloc[:150], y_train=y.iloc[:150], problem_type="time series regression",
                     problem_configuration={"gap": 2, "max_delay": 7, "forecast_horizon": 7, "date_index": "date"},
                     allowed_model_families=["linear_model"])
aml.search()

aml.best_pipeline.fit(X.iloc[:150], y.iloc[:150])
aml.best_pipeline.score(X.iloc[150:], y.iloc[150:], ['R2'], X.iloc[:150], y.iloc[:150])['R2']
==============================================
ValueError                                Traceback (most recent call last)
/var/folders/tp/38kwkh8d18l7ylrtm232hgmd_rd2bk/T/ipykernel_6218/2552366871.py in <module>
      1 search.best_pipeline.fit(X.iloc[:150], y.iloc[:150])
----> 2 search.best_pipeline.score(X.iloc[150:], y.iloc[150:], ['R2'], X.iloc[:150], y.iloc[:150])['R2']

~/PycharmProjects/evalml/evalml/pipelines/time_series_regression_pipeline.py in score(self, X, y, objectives, X_train, y_train)
     38         X_train, y_train = self._convert_to_woodwork(X_train, y_train)
     39         objectives = self.create_objectives(objectives)
---> 40         y_predicted = self.predict_in_sample(X, y, X_train, y_train)
     41         return self._score_all_objectives(
     42             X, y, y_predicted, y_pred_proba=None, objectives=objectives

~/PycharmProjects/evalml/evalml/pipelines/time_series_pipeline_base.py in predict_in_sample(self, X, y, X_train, y_train, objective)
    187         target = infer_feature_types(y)
    188         features = self.compute_estimator_features(X, target, X_train, y_train)
--> 189         predictions = self._estimator_predict(features, target)
    190         predictions.index = y.index
    191         predictions = self.inverse_transform(predictions)

~/PycharmProjects/evalml/evalml/pipelines/time_series_pipeline_base.py in _estimator_predict(self, features, y)
    241         if self.estimator.predict_uses_y:
    242             y_arg = y
--> 243         return self.estimator.predict(features, y=y_arg)

~/PycharmProjects/evalml/evalml/pipelines/components/component_base_meta.py in _check_for_fit(self, X, y)
     27                 return method(self)
     28             elif y is None:
---> 29                 return method(self, X)
     30             else:
     31                 return method(self, X, y)

~/PycharmProjects/evalml/evalml/pipelines/components/estimators/estimator.py in predict(self, X)
     77                 X.columns = [x for x in X.columns]
     78             print(self._component_obj)
---> 79             predictions = self._component_obj.predict(X)
     80         except AttributeError:
     81             raise MethodPropertyNotFoundError(

~/miniconda3/envs/main3.8/lib/python3.8/site-packages/sklearn/linear_model/_base.py in predict(self, X)
    236             Returns predicted values.
    237         """
--> 238         return self._decision_function(X)
    239 
    240     _preprocess_data = staticmethod(_preprocess_data)

...

~/miniconda3/envs/main3.8/lib/python3.8/site-packages/sklearn/utils/validation.py in _assert_all_finite(X, allow_nan, msg_dtype)
    101                 not allow_nan and not np.isfinite(X).all()):
    102             type_err = 'infinity' if allow_nan else 'NaN, infinity'
--> 103             raise ValueError(
    104                     msg_err.format
    105                     (type_err,

ValueError: Input contains NaN, infinity or a value too large for dtype('float64').
```
This issue doesn't seem to occur with other model families.

**Comment by freddyaboulton** [comment]
@ParthivNaresh I think the problem is that in your repro `y` has 201 entries while `X` has 200 entries. This means that `y.iloc[150:]` and `X.iloc[150:]` have 51 vs 50 entries, respectively. 

I wouldn't expect us to be able to score successfully in that case. In fact, changing the problem type to `regression` in this repro would also fail for any model family.

I think what you found about this bug only affecting elastic net is interesting. I can get a same bug if I use another scikit-learn based model family, like random forest or decision tree, but not xgboost or catboost. I think those estimators are actually able to ignore NaNs in the input! 

So in summary, I think we should close.
